lib/cc/cli/version_checker.rb

```diff
@@ -0,0 +1,83 @@
+require "uuid"
+
+module CC
+  module CLI
+    class VersionChecker
+      include CC::CLI::Output
+
+      VERSION_CHECK_TIMEOUT = 60 * 60 # 1 Hour in seconds
+      DEFAULT_VERSIONS_URL = "https://versions.codeclimate.com".freeze
+
+      def check
+        return unless check_version?
+
+        print_new_version_message if outdated?
+      end
+
+      private
+
+      def check_version?
+        first_arg = ARGV.first
+        if %w[--check-version].include? first_arg
+          ARGV.shift
```

Comment:
I think it's mixing responsibilities to have this class in charge of argument parsing. I would prefer this class just check the version and have the caller (`Runner`) invoke it or not based on the arguments (it's already got that responsibility).

```rb
class Runner
  def run
    VersionChecker.run if check_version?

```